libpromises/evalfunction.c

```diff
@@ -6990,6 +6990,10 @@ void ModuleProtocol(EvalContext *ctx, char *command, const char *line, int print
             StringSetAddSplit(tags, content, ',');
             StringSetAdd(tags, xstrdup("source=module"));
         }
+        else if (1 == sscanf(line + 1, "persistence=%ld", persistence))
```

Comment:
Could you make the necessary change please?  I think it will be much more efficient; you can just amend my commit or add another one on top.  Otherwise we may need to go back and forth again and again... I don't have Solaris running anywhere anymore :)